python/tests/api/pyspark/experimental/test_profiler_function.py

```diff
@@ -43,6 +68,46 @@ def input_df(self, test_columns):
         )
         return input_df
 
+    def test_puzzle(self, input_df):
+        # TODO: consider a constant for this, or better encapsulation.
+        cp = "col_name string, col_profile binary"
+
+        input_df.show()
+
+        def profiler_nested(pdf_iterator: Iterable[pd.DataFrame]) -> Iterable[pd.DataFrame]:
+            for input_df in pdf_iterator:
+                res = why.log(input_df)
+                res_df = pd.DataFrame(columns=[COL_NAME_FIELD, COL_PROFILE_FIELD])
+                for col_name, col_profile in res.view().get_columns().items():
+                    d = {
+                        COL_NAME_FIELD: [col_name],
+                        COL_PROFILE_FIELD: [col_profile.to_protobuf().SerializeToString()],
+                    }
+                    df_temp = pd.DataFrame(data=d)
+                    res_df = res_df.append(df_temp)
+                yield res_df
+
+        def profiler_wrapped(pdf_iterator: Iterable[pd.DataFrame]) -> Iterable[pd.DataFrame]:
+            return whylogs_pandas_map_profiler(pdf_iterator)
+
+        # profile_bytes_df = input_df.mapInPandas(whylogs_pandas_map_profiler, schema=cp)  # crashes?
+        # profile_bytes_df = input_df.mapInPandas(profiler_local, schema=cp)  # crashes?
+        # profile_bytes_df = input_df.mapInPandas(profiler_nested, schema=cp)  # works?
+        profile_bytes_df = input_df.mapInPandas(profiler_wrapped, schema=cp)  # works?
```

Comment:
Why did this pass and the nested but not the others do you think?

Reply:
My best guess at the moment is some weird interaction between Spark and Docker. The "normal" code seems to work after one of these work-arounds runs, so maybe something like a race condition?